helm/charts/infra/templates/connector/secret.yaml

```diff
@@ -1,7 +1,5 @@
 
 {{- if include "connector.enabled" . | eq "true" }}
-{{- $accessKey := default "" .Values.connector.config.accessKey -}}
-{{- if or (not $accessKey) (and (not (hasPrefix "file:" $accessKey)) (not (hasPrefix "env:" $accessKey))) }}
```

Comment:
I believe we want to create this secret in all cases now

Reply:
This ignores the case when a user creates the access key secret out of band and want to load an existing secret rather than creating on through Helm. One way of addressing that could be to use a different, e.g. `accessKeySecret`.

Reply:
We also need to consider the migration for users who may be using `env:` or `file:` vallues. In both cases, the access key secret should not be rendered. `NOTES.txt` should also be updated to warn users about the configuration change.

Reply:
> This ignores the case when a user creates the access key secret out of band and want to load an existing secret rather than creating on through Helm

It would be really great to start simplifying the helm charts and being a lot more opinionated about what we support.

Using an existing secret seems like a reasonable thing to support, but do we really need to allow them to call it anything? If we require them to name the secret according to our convention, then I think this code still works, because we look up the value from the existing secret.

Is there a use case that requires us to support an arbitrarily named secret?

Reply:
The chart should not modify anything it does not own. In the case user provides an existing secret for the access key, the secret template should see it as an externally configured secret and not produce nothing.

Failing that, the best case scenario would to fail cause Helm cannot modify the resources it does not own.

The worst case is Helm takes ownership of the resource and overrides the contents with its own values. The is problematic because we don't know the origin of the resource. It could be created by something like `external-secrets` where the real value lives in some external secret store. 

Taking ownership also means the resource will subject it to its lifecycle rules. This usually means deleting the resource when the chart is uninstalled. One of the reasons why users may want to provide their own Kubernetes secret is to keep it away from Helm's lifecycle management.

Reply:
We chatted offline and I think we have a good way to solve this now:

1. If the access key is specified from the command line (`--set connector.accessKey=<acessKeyValue>`) or from a `values.yaml` 
   - create the kubernetes secret
   - set it in the deployment using `envFrom` (instead of from the configmap like we do now)
2. If the access key is not specified, we can assume the value is coming from anexisting kubernetes secret that is not managed by our chart
   - the operator will have to specify the `envFrom` using a `values.yaml`
3. If the operator is using a CSI driver to specify a secret (ex: [aws secrets manager](https://docs.aws.amazon.com/secretsmanager/latest/userguide/integrating_csi_driver.html), they will need to set a `volume`, `volumeMount`, and an env var like `INFRA_ACCESS_KEY=/path/to/secret/mount/file`

We're going to wait until we split out the helm charts for the server and connector to make this change.

Reply:
I've updated this PR to make the change backwards compatible, and removed most of the helm chart changes.